app/src/main/java/org/simple/clinic/home/overdue/removepatient/RemoveAppointmentSheetController.kt

```diff
@@ -0,0 +1,65 @@
+package org.simple.clinic.home.overdue.removepatient
+
+import io.reactivex.Observable
+import io.reactivex.ObservableSource
+import io.reactivex.ObservableTransformer
+import io.reactivex.rxkotlin.ofType
+import io.reactivex.rxkotlin.withLatestFrom
+import org.simple.clinic.ReportAnalyticsEvents
+import org.simple.clinic.overdue.AppointmentRepository
+import org.simple.clinic.widgets.UiEvent
+import javax.inject.Inject
+
+typealias Ui = RemoveAppointmentSheet
+typealias UiChange = (Ui) -> Unit
+
+class RemoveAppointmentSheetController @Inject constructor(
+    private val repository: AppointmentRepository
+) : ObservableTransformer<UiEvent, UiChange> {
+
+  override fun apply(event: Observable<UiEvent>): ObservableSource<UiChange> {
+    val replayedEvents = event.compose(ReportAnalyticsEvents()).replay().refCount()
+
+    return Observable.merge(
+        reasonClicks(replayedEvents),
+        doneClicks(replayedEvents))
+  }
+
+  private fun reasonClicks(events: Observable<UiEvent>): Observable<UiChange> {
+    val mergedReasons = Observable.merge(
+        events.ofType<AlreadyVisitedReasonClicked>(),
+        events.ofType<CancelReasonClicked>())
+
+    return mergedReasons
+        .distinctUntilChanged()
+        .map { { ui: Ui -> ui.enableDoneButton() } }
```

Comment:
I'm nitpicking here, but I think explicitly declaring the merged streams is a bit more understandable than `mergedStreams`:

```kotlin
val alreadyVisitedClicks = events.ofType<AlreadyVisitedReasonClicked>()
val cancelClicks = events.ofType<CancelReasonClicked>()

return Observable.merge(alreadyVisitedClicks, cancelClicks)
    .distinctUntilChanged()
    .map { { ui: Ui -> ui.enableDoneButton() } }
```

This is just a suggestion. Feel free to not make any changes here.